### ISSUE TITLE
Expand intersect is not working

### ISSUE THREAD

**Issue by janjilek** [orientechnologies/orientdb#5121]
Hello, i am using version 2.1.4 and getting empty result on following query:

```sql
SELECT expand(intersect($q1, $q2))
LET q1 = (SELECT FROM Product WHERE title LUCENE 'ham*'),
q2 = (SELECT expand(IN('Belong')[@class = Product]) FROM Category WHERE type = "ebooks")
```

Strange is, that without expand I can see the list of records in intersect property.

**Comment by janjilek** [comment]
Walk-around for this issue is not to use expand, but use intersect with the fetchplan:

```sql
SELECT intersect($neco, $neco2) as res
LET neco2 = (SELECT FROM Product WHERE title LUCENE 'ham*'),
neco = (SELECT expand(IN('Belong')[@class = Product]) FROM Category WHERE type = "ebooks") 
fetchplan res:1
```

**Comment by janjilek** [comment]
Any suggestion?

**Comment by luigidellaquila** [comment]
Hi @janjilek 

it seems the ```intersect``` is not returning a proper object, so that expand can manage it.
Could you please try this and see if it returns the expected results?

```sql
select expand(foo) from (
SELECT intersect($q1, $q2) as foo
LET q1 = (SELECT FROM Product WHERE title LUCENE 'ham*'),
q2 = (SELECT expand(IN('Belong')[@class = Product]) FROM Category WHERE type = "ebooks")
)
```

In the meantime, I'm flagging it as a bug

Thanks

Luigi

**Comment by janjilek** [comment]
Looks, that returns expected result!  

Thanks
Jan

**Comment by luigidellaquila** [comment]
Fixed in both 2.1.x and develop branches.
Closing

Thanks
